expire-old-content.js

```
@@ -0,0 +1,233 @@
+        content = fs.readFileSync(componentPath, 'utf8');
+    } catch (error) {
+        console.error(`Error reading component file ${componentPath}:`, error);
+        return false;
+    }
+    let hasChanges = false;
+    let removedImages = new Set();
+
+    // First pass: collect expired image names that should be removed
+    content.replace(/<img [^>]*src=\{([^}]+)\}[^>]*valid-until="([^"]+)"[^>]*>/g, (match, imgVar, dateStr) => {
+        const expirationDate = new Date(dateStr);
+        if (isNaN(expirationDate.getTime())) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)